osgp/protocol-adapter-oslp/web-device-simulator/src/main/java/org/opensmartgridplatform/webdevicesimulator/service/OslpChannelHandler.java

```diff
@@ -1200,8 +1117,14 @@ private void handleSetEventNotificationsRequest(final Device device, final SetEv
     private void handleUpdateFirmwareRequest(final Device device, final UpdateFirmwareRequest request) {
         LOGGER.debug("handle UpdateFirmwareRequest for device: {}, with serialized size of {}",
                 device.getDeviceIdentification(), request.getSerializedSize());
-        // For now, do nothing, perhaps store firmware version, so that it can
-        // be displayed ???
+
+        this.sendDelayedDeviceRegistration(device);
+
+        // Send a firmware activation event after the device registration has
+        // (likely) been finished
+        final Oslp.Event event = Oslp.Event.FIRMWARE_EVENTS_ACTIVATING;
+        final String description = "A new firmware is activated";
+        this.sendEventWithCustomDelay(device, event, description, 7000);
```

Comment:
These additional steps for `handleUpdateFirmwareRequest()` are the only real update for the `OslpChannelHandler` class.  All other changes seem formatting related.

Reply:
Ha, the answer to something I already noticed. I left some comments in other parts of the code as well.
Since they apparently only concern changes due to reformatting, doing something about them can be considered optional (use your own judgement).